### ISSUE TITLE
Add support for Gradle

### ISSUE THREAD

**Issue by chicxurug** [FrontEndART/AIFix4SecCode#44]
Implement a GradlePatchCompiler class that extends the GenericPatchCompiler class and adds support for compiling gradle projects.

**Comment by searchlab-team** [comment]
Is there any gradle test project to try out GradlePatchCompiler class?

**Comment by searchlab-team** [comment]
What gradle project should I look for to try out the GradlePatchCompiler? For example there is the getVulnerabilityLocations method in OpenStaticAnalyzer.java and if I'm not wrong, it looks for supported warning types in the project. I found one gradle project that the SpotBugs gave EI_EXPOSE_REP2 warning and I see that the  SUPPORTED_PROBLEM_TYPES contains this type as well. I tried out with this gradle project, but the getVulnerabilityLocations didn't find any supported problem type.

**Comment by chicxurug** [comment]
You can use this one: https://github.com/jankotek/mapdb
If the SpotBugs version do not match, you might not find an issue. We tested this project with the OSA toolchain and it should contain fixable warnings.

**Comment by searchlab-team** [comment]
I tried with this project, but still finds no vulnerabilityLocation. This was a problem before with the maven test-project, but building it, solved the problem, so I tried building this gradle project as well, but still finds nothing.
My JDK version is 11, Spotbugs version is 4.5.3, Gradle version 7.4.1, but with this version there were some issues building gradle project so I used `gradlew --version`  command wich says now that the gradle version is 5.6.4
It might be my Spotbugs version, what version should I use?
